web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -1457,33 +1491,18 @@ fixmystreet.display = {
             width: width,
             height: height
         });
-        $('#try_again').click(function(e){
+
+        $('.mobile-map-banner span').text(translation_strings.right_place);
+
+        $('#problems_nearby').click(function(e){
             e.preventDefault();
             history.back();
         });
 
-        $('.mobile-map-banner span').text(translation_strings.right_place);
+        $('#try_again').click(fixmystreet.mobile_reporting.try_again);
 
         // mobile user clicks 'ok' on map
-        $('#mob_ok').toggle(function(){
-            //scroll the height of the map box instead of the offset
-            //of the #side-form or whatever as we will probably want
-            //to do this on other pages where #side-form might not be
-            $('html, body').animate({ scrollTop: height-60 }, 1000, function(){
-                $('html').removeClass('only-map');
-                $('#mob_sub_map_links').addClass('map_complete');
-                $('#mob_ok').text(translation_strings.map);
-            });
-        }, function(){
-            $('html, body').animate({ scrollTop: 0 }, 1000, function(){
-                $('html').addClass('only-map');
-                $('#mob_sub_map_links').removeClass('map_complete');
-                $('#mob_ok').text(translation_strings.ok);
-                if (fixmystreet.duplicates) {
-                    fixmystreet.duplicates.hide();
-                }
-            });
-        });
+        $('#mob_ok').toggle(fixmystreet.mobile_reporting.confirmed_location, fixmystreet.mobile_reporting.try_again);
```

Comment:
There's something a bit odd about this button on iOS - sometimes it needs two clicks to activate it. The bug appears consistently when half-scrolling up to the map then clicking the "Try again" button; but generally doesn't happen when clicking the "Map" button with the chevron:



![try_again_75_full](https://user-images.githubusercontent.com/4776/73364790-1228fd00-42a3-11ea-9294-fcd57a2e61a4.gif)

Reply:
Steps to reproduce (on iOS):

1. Enter a location on front page
1. Tap the map to place a pin
1. Tap "OK"
1. Wait for "Report your problem" form to scroll in to view
1. Scroll up until the "Try again" button is visible on map
1. Tap "Try again" button
1. Wait for map to scroll into view and overlay to disappear
1. (optional, doesn't seem to affect bug incidence) Tap map to reposition report.
1. Tap "OK" button
1. Notice that nothing happens, apart from the "OK" button becoming highlighted.
1. Tap "OK" button again
1. "Report your problem" form scrolls in to view as expected.

These steps work as expected:

1. Enter a location on front page
1. Tap the map to place a pin
1. Tap "OK"
1. Wait for "Report your problem" form to scroll in to view
1. Scroll up until the "Map" button is visible
1. Tap "Map" button
1. Wait for map to scroll into view and overlay to disappear
1. (optional, doesn't seem to affect bug incidence) Tap map to reposition report.
1. Tap "OK" button
1. "Report your problem" form scrolls in to view as expected.

Is there some kind of `mouseenter` event or `:hover` CSS class on the "OK" button perhaps. Because the "OK" button is the same DOM element as the "Map" button, I wonder if tapping it to reveal the map counts as the first tap which highlights the button, and the second activates it. Clicking the "Try again" button doesn't activate the hover state on the "Map"/"OK" button, hence two taps required.